packages/core-components-next/src/components/SelectElement/index.vue

```diff
@@ -0,0 +1,280 @@
+    'createdAt',
+  ]) as ContentContainer[];
+  return flatMap(containers, (it) => [it, ...getSubcontainers(it)]);
+});
+
+const getContainerTypes = (type: string) => {
+  return map(schemaService.getSupportedContainers(type), 'type');
+};
+
+const getTypePosition = ({ type }: { type: string }) => {
+  return rootContainerTypes.value.indexOf(type);
+};
+
+const isRootContainer = ({ parentId, type }: Activity) => {
+  if (!selection.activity) return false;
+  return (
+    parentId === selection.activity.id &&
+    rootContainerTypes.value.includes(type)
+  );
+};
+
+const getSubcontainers = (container: ContentContainer) => {
+  const { getDescendants } = activityUtils;
+  return sortBy(getDescendants(items.activities, container), 'position');
+};
+
+const showActivityElements = async (activity: Activity) => {
+  selection.activity = activity;
+  const elements: ContentElement[] = await fetchElements(
+    processedContainers.value,
+  );
+  items.contentContainers = processedContainers.value.map((container) => {
+    return assignElements(container, activity, elements);
+  });
+};
+
+const assignElements = (
+  container: ContentContainer,
+  activity: Activity,
+  elements: ContentElement[],
+) => {
+  const containerElements = elements
+    .filter(({ activityId }) => activityId === container.id)
+    .map((element) => ({ ...element, activity }));
+  return { ...container, elements: sortBy(containerElements, 'position') };
+};
+
+const toggleElementSelection = (element: ContentElement) => {
+  const { elements } = selection;
+  const existing = elements.find((it) => it.id === element.id);
+  selection.elements = existing
+    ? elements.filter((it) => it.id !== element.id)
+    : elements.concat(element);
+};
+
+const toggleSelectAll = () => {
+  selection.elements = allElementsSelected.value ? [] : elements.value;
+};
+
+const deselectActivity = () => {
+  selection.activity = undefined;
+  items.contentContainers = [];
+  selection.elements = [...props.selected];
+};
+
+const selectRepository = async (repository: Repository) => {
+  selection.repository = repository;
+  deselectActivity();
+  const isCurrentRepository = currentRepository.id === repository.id;
+  items.activities = isCurrentRepository
+    ? currentRepository.activities
+    : await fetchActivities(repository);
+};
+
+const fetchActivities = loader(async function (repository: Repository) {
+  return api.fetchActivities(repository.id);
+}, 500);
+
+const fetchElements = loader(async function (containers: ContentContainer[]) {
+  const repositoryId = selection.repository?.id;
+  const params = { ids: map(containers, 'id') };
+  return api.fetchContentElements(repositoryId, params);
+}, 500);
+
+const save = () => {
+  emit('selected', [...selection.elements]);
+  close();
+};
+
+const close = () => emit('close');
+
+const openInEditor = (elementId: number) => {
+  const { repository, activity } = selection;
+  if (!repository || !activity) return;
+  const url = new URL(
```

Comment:
Should we hoist this up and leave it to system consuming the component to implement